src/cbmc/cbmc_parse_options.cpp

```diff
@@ -516,6 +516,10 @@ int cbmc_parse_optionst::doit()
   if(get_goto_program_ret!=-1)
     return get_goto_program_ret;
 
+  INVARIANT(
+    !goto_model.check_internal_invariants(*this),
+    "goto program internal invariants failed");
+
```

Comment:
Not entirely sure what the best approach is, but this might be relevant going forward as we extend this.
I'm considering calling my goto_program checks from the try block of cbmc_parse_optionst::get_goto_program so that any exceptions of type 'incorrect_goto_program_exceptiont' are caught there.
If called directly from the doit() method exceptions are then caught in parse_options_baset::main()
which has only the base exception type or the invalid_user_input_exceptiont type. 
(I think)

Reply:
Well in #3123 @danpoe makes it switchable to throw exception of check invariant. Also, the whole well-formedness is optional (user-wise). So, yes, I expect the point of entry to be modified at some point.

Reply:
Ok.